python/pyspark/sql/session.py

```diff
@@ -543,9 +543,12 @@ def getOrCreate(self) -> "SparkSession":
 
                 session = SparkSession._instantiatedSession
                 if session is None or session._sc._jsc is None:
+                    SparkContext._ensure_initialized()
```

Comment:
nit: As we initiate `SparkSession` above, I guess `SparkContext` is initiated too, do we need to ensure it again here?

Reply:
Thank you for review, @viirya . Yes, this is required before we need to read `spark.submit.appName` from JVM.

Without this, PySpark's `SparkConf` only is holding Python-side setting.

Reply:
Hmmm .. actually I think this causes SPARK-50281 by not passing the core confs like `spark.jars` because `_ensure_initialized` creates JVM but the jars have to be passed when launching Py4J server.

Reply:
It has to be set through `SparkContext.getOrCreate` -> `SparkContext.__init__` -> `SparkContext._ensure_initialized(self, gateway=gateway, conf=conf)` -> `Py4J`